pkg/utils.py

```diff
@@ -299,37 +321,46 @@ def json_to_junit(test_name, data_json, metrics_config):
         testsuites, "testsuite", name=f"{test_name} nightly compare"
     )
     failures_count = 0
-    test_count=0
-    for run in data_json:
-        run_data = {
-            str(key): str(value).lower()
-            for key, value in run.items()
-            if key in ["timestamp"]
-        }
-        for metric, value in run["metrics"].items():
-            test_count+=1
-            failure = "false"
-            if not value["percentage_change"] == 0:
-                failure = "true"
-                failures_count += 1
-            labels = metrics_config[metric]["labels"]
-            label_string = " ".join(labels) if labels else ""
-            testcase = ET.SubElement(
-                testsuite,
-                "testcase",
-                name=f"{label_string} {metric} regression detection",
-                attrib=run_data,
-            )
-            if failure == "true":
-                failure_element = ET.SubElement(testcase, "failure")
-                failure_element.text = f"{metric} has a value of {value['value']:.2f} with a percentage change of {value['percentage_change']:.2f}% over the previous runs"
+    test_count = 0
+    for metric, value in metrics_config.items():
+        test_count += 1
+        labels = value["labels"]
+        label_string = " ".join(labels) if labels else ""
+        testcase = ET.SubElement(testsuite, "testcase", name=f"{label_string} {metric} regression detection", timestamp=str(int(datetime.now().timestamp())))
+        if [run for run in data_json if not run["metrics"][metric]["percentage_change"] == 0]:
+            failures_count +=1
+            failure = ET.SubElement(testcase,"failure")
+            failure.text = "\n"+generate_tabular_output(data_json, metric_name=metric)+"\n"
+
     testsuite.set("failures", str(failures_count))
     testsuite.set("tests", str(test_count))
     xml_str = ET.tostring(testsuites, encoding="utf8", method="xml").decode()
     dom = xml.dom.minidom.parseString(xml_str)
     pretty_xml_as_string = dom.toprettyxml()
     return pretty_xml_as_string
 
+def generate_tabular_output(data: dict, metric_name: str) -> str:
```

Comment:
I think we should just present in the XML a csv -- uuid, buildURL, metric, value, change-point[true|false] not sure how the XML file will read in the table view.